certbot/certbot/_internal/main.py

```diff
@@ -130,7 +130,37 @@ def _get_and_save_cert(le_client, config, domains=None, certname=None, lineage=N
     return lineage
 
 
-def _handle_subset_cert_request(config, domains, cert):
+def _handle_unexpected_key_type_migration(config, cert):
+    # type: (configuration.NamespaceConfig, storage.RenewableCert) -> None
+    """
+    This function ensures that the user will not implicitly migrate an existing key
+    from one type to another in the situation where a certificate for that lineage
+    already exist and they have not provided explicitly --key-type and --cert-name.
+    :param config: Current configuration provided by the client
+    :param cert: Matching certificate that could be renewed
+    """
+    if (config.verb in ["certonly", "run"]
```

Comment:
I don't think we need this `config.verb` check do we? I think we can only get here if effectively running "certonly" or "run".

If for some reason we want to keep this check, I think we also need to check for aliases like [`auth`](https://github.com/certbot/certbot/blob/db2ffea351c78af1bf682cda64fc58dd91ab3391/certbot/certbot/_internal/cli/helpful.py#L54) and [`everything`](https://github.com/certbot/certbot/blob/db2ffea351c78af1bf682cda64fc58dd91ab3391/certbot/certbot/_internal/cli/helpful.py#L65) and hope we don't add new aliases that need to be checked here too in the future.

Reply:
My initial concerns was to build a function for which all controls required on the execution context to decide are done in the function itself. So that if the function is called in another context, it is a noop. I did that because it is called in several places in the code, and the `config.verb` is actually decided several layers up, making it more difficult for the reader to understand in which context we are precisely.

Now if this harms the maintainability because potential new aliases could be used, I will remove it. The unit tests will catch easier the fact that the new function is added in another process (test break, mock is required) than the fact that an untested alias silently migrates key types.

Reply:
Check removed.